modules/bibauthorid/lib/bibauthorid_dbinterface.py

```diff
@@ -3737,11 +3737,11 @@ def get_name_variants_for_authors(authors):  # get_indexable_name_personids
     @return: real author name and the author identifiers which carry that name ((name, pids),)
     @rtype: tuple ((str, bytes),)
     '''
-    name_variants = run_sql("""select id, personids
-                               from aidDENSEINDEX
-                               where id in %s
-                               and flag=1"""
-                            % _get_sqlstr_from_set(authors))
+
+    name_variants = run_sql("""select id, personids from aidDENSEINDEX
+                               where%s flag=1"""
+                            % (" id in " + _get_sqlstr_from_set(authors) + " and"
+                               if len(authors) else ""))
```

Comment:
Two cases:
`select id, personids from aidDENSEINDEX where flag=1`
or
`select id, personids from aidDENSEINDEX where id in ? and flag=1`